File: LeetCode/Graph/BFS/JS/Leetcode513-find-bottom-left-tree-value.js
var findBottomLeftValue = function (root) {
    let [hei, val] = [0, 0];
    let dfs = function (node, h) {
        if (!node) return;
        h++;
        dfs(node.left, h);
        dfs(node.right, h)
        if (h > hei) {
            hei = h;
            val = node.val;
        }
    }
    dfs(root, 0);
    return val;
};

File: LeetCode/Graph/BFS/Golang/Leetcode513-find-bottom-left-tree-value.go
package sbgo

func findBottomLeftValue(root *TreeNode) int {
	val, hei := 0, 0
	var dfs func(node *TreeNode, h int)
	dfs = func(node *TreeNode, h int) {
		if node == nil {
			return
		}
		h++
		dfs(node.Left, h)
		dfs(node.Right, h)
		if h > hei {
			hei = h
			val = node.Val
		}
	}
	dfs(root, 0)
	return val
}
